src/analyzer/engines/gensim.py

```diff
@@ -0,0 +1,71 @@
+from src.analyzer.engines.engine import ModelingEngine
+
+from gensim.models.coherencemodel import CoherenceModel
+from gensim.models.ldamodel import LdaModel
+from gensim.models.tfidfmodel import TfidfModel
+from gensim.models import Nmf
+
+class GensimEngine(ModelingEngine):
+    """
+    GensimEngine implements the methods for fitting and evaluating gensim models on a
+    corpus.
+    """
+    def __init__(self, min_topics, max_topics):
+        self.min_topics = min_topics
+        self.max_topics = max_topics
+        self.models = {}
+        self.dictionary = None
+        self.type = ""
+        self.corpus = None
+
+    def fit(self, model, dictionary, corpus, **kwargs):
```

Comment:
Since it looks like we expect users to call `fit` directly, I think it would be nice to make these keyword arguments rather than positions (I always have a really hard time remember what order to put positional arg in). Maybe `def fit(self, model="lda", dictionary=None, corpus=None, **kwargs)`

Also, let's remember to add the docstring here so that the user will know what format is required for the `dictionary` and `corpus` (will the `corpus` data structure be the same for Gensim and sklearn?). If either of these is `None`, we might want to return an error that tells the user to run `OnlineTextCorpus(...).get_dictionary()`.